api/user.go

```diff
@@ -27,13 +29,34 @@ import (
 // UserAPI handles request to /api/users/{}
 type UserAPI struct {
 	BaseAPI
-	currentUserID int
-	userID        int
+	currentUserID    int
+	userID           int
+	SelfRegistration bool
+	IsAdmin          bool
+	AuthMode         string
 }
 
 // Prepare validates the URL and parms
 func (ua *UserAPI) Prepare() {
 
+	authMode := strings.ToLower(os.Getenv("AUTH_MODE"))
+	if authMode == "" {
+		authMode = "db_auth"
+	}
+	ua.AuthMode = authMode
+
+	selfRegistration := strings.ToLower(os.Getenv("SELF_REGISTRATION"))
+	if selfRegistration == "on" {
+		ua.SelfRegistration = true
+	}
+
+	if ua.Ctx.Input.IsPost() {
+		sessionUserID := ua.GetSession("userId")
+		if sessionUserID == nil {
+			return
+		}
+	}
+
 	ua.currentUserID = ua.ValidateUser()
```

Comment:
I think this should be differentiated by logged in and not logged in status as for adding sign up process.